libsql/src/database.rs

```diff
@@ -358,11 +378,31 @@ impl Database {
             }
 
             #[cfg(feature = "replication")]
-            DbType::Sync { db } => {
+            DbType::Sync { db, encryption_key } => {
                 use crate::local::impls::LibsqlConnection;
 
                 let conn = db.connect()?;
 
+                if !cfg!(feature = "encryption-at-rest") {
+                    let _ = encryption_key;
+                    tracing::warn!("Encryption at rest is not enabled, ignoring encryption_key");
```

Comment:
is this actually worth a warn? I think this should maybe be a debug at most.

Reply:
actually, I think it's the other way round and it should be a runtime error, we don't want users accidentally not encrypting their data on disk